### ISSUE TITLE
error line 200, in from_pretrained  assert len(keys) == len(sd)

### ISSUE THREAD

**Issue by Sandy4321** [karpathy/minGPT#128]
error line 200, in from_pretrained  assert len(keys) == len(sd)

len(keys)
581
len(sd)
629

![image](https://github.com/karpathy/minGPT/assets/11426119/213e704f-e387-4cee-b0d0-2434036a86c9)

mingpt-master\mingpt-master\mingpt\model.py", line 200, in from_pretrained
  assert len(keys) == len(sd)

windows 11 OS
torch      version                    1.13.1
transformers           version                            4.36.2

**Comment by Sandy4321** [comment]
Python 3.10.11

**Comment by Sandy4321** [comment]
code
import torch
from transformers import GPT2Tokenizer, GPT2LMHeadModel
from mingpt.model import GPT
from mingpt.utils import set_seed
from mingpt.bpe import BPETokenizer
set_seed(3407)

use_mingpt = True # use minGPT or huggingface/transformers model?
model_type = 'gpt2-xl'
device = 'cuda'
if use_mingpt:
    model = GPT.from_pretrained(model_type)
else:
    model = GPT2LMHeadModel.from_pretrained(model_type)
    model.config.pad_token_id = model.config.eos_token_id # suppress a warning

# ship model to device and set to eval mode
model.to(device)
model.eval();

q = 0

**Comment by Sandy4321** [comment]
mingpt-master\mingpt-master\s_mingpt_jan21.py", line 15, in <module>
  model = GPT.from_pretrained(model_type)
File "c:\mingpt-master\mingpt-master\mingpt\model.py", line 200, in from_pretrained
  assert len(keys) == len(sd)

builtins.AssertionError

**Comment by xiaoyuxie-vico** [comment]
I have the same issue.

**Comment by Sandy4321** [comment]
MAY @karpathy  FIX IT ?

**Comment by chenokay** [comment]
is it fixed?

**Comment by vanduc103** [comment]
It was fixed in here #120 and also the reason of the error.
